herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java

```diff
@@ -386,6 +387,21 @@ public void recovery(
                     + ": Local data is absent, and actual ledger list " + this.actualLedgersList.getActiveLedgers()
                     + " does not contain first ledger of ever: " + this.actualLedgersList.getFirstLedger()));
         }
+        for (long ledgerId : actualLedgersList.getActiveLedgers()) {
+            try {
+                bookKeeper.getLedgerManager().readLedgerMetadata(ledgerId).get();
```

Comment:
is use use org.apache.bookkeeper....FutureUtils.result() you won't have to deal with "ExecutionException"